app/models/notify_user/base_notification.rb

```diff
@@ -211,190 +51,97 @@ def delay_time(options)
+    def target_has_unsubscribed?(channel_name=nil)
+      return Unsubscribe.has_unsubscribed_from?(target, type, group_id, channel_name)
     end
 
-    #deliver to specific channel methods
-
-    # Deliver a single notification to a specific channel.
-    def self.deliver_notification_channel(notification_id, channel_name)
-      notification = self.find(notification_id) # Raise an exception if not found.
-
-      channel_options = channels[channel_name.to_sym]
-      channel = (channel_name.to_s + "_channel").camelize.constantize
-
-      unless notification.user_has_unsubscribed?(channel_name)
-        channel.deliver(notification, channel_options)
-      end
+    def read?
+      read_at.present?
     end
 
-    # Deliver a aggregated notifications to a specific channel.
-    def self.deliver_notifications_channel(notifications, channel_name)
-      channel_options = channels[channel_name.to_sym]
-      channel = (channel_name.to_s + "_channel").camelize.constantize
-
-      #check if user unsubsribed from channel type
-      unless notifications.first.user_has_unsubscribed?(channel_name)
-        channel.deliver_aggregated(notifications, channel_options)
-      end
+    def mark_as_read!
+      self.read_at = Time.zone.now
+      self.save
     end
 
-    # Prepares a single channel for aggregation
-    def self.notify_aggregated_channel(notification_id, channel_name)
-      notification = self.find(notification_id) # Raise an exception if not found.
-
-      # Find any pending notifications with the same type and target, which can all be sent in one message.
-      if self.aggregate_grouping
-        notifications = self.pending_aggregation_by_group_with(notification)
-      else
-        notifications = self.pending_aggregation_with(notification)
-      end
+    def parents_in_group
+      return self.class.none unless self.aggregate_grouping
+      self.class.for_target(target)
+        .where(parent_id: nil)
+        .where(group_id: group_id)
+    end
 
-      notifications.map(&:mark_as_sent)
-      notifications.map(&:save)
+    # Scopes:
+    def self.for_target(target)
+      where(target: target)
+    end
 
-      # If the notification has been marked as read before it's sent we don't want to send it.
-      return if notification.read? || notifications.empty?
+    # Get the unread count for a given target:
+    def self.unread_count_for_target(target)
+      for_target(target)
+        .where('parent_id IS NULL')
+        .where('read_at IS NULL')
+        .count
+    end
 
-      if notifications.length == 1
-        # Despite waiting for more to aggregate, we only got one in the end.
-        self.deliver_notification_channel(notifications.first.id, channel_name)
-      else
-        # We got several notifications while waiting, send them aggregated.
-        self.deliver_notifications_channel(notifications, channel_name)
-      end
+    # Configure a channel
+    def self.channel(name, options={})
```

Comment:
We spoke about maybe renaming this method to `add_channel` or `register_channel`, but, I guess that's a breaking change for no real reason